File: 操作系统/Windows API/Windows 程序设计/第6版/CPlusPlus/Chapter17/CustomizableClassHierarchy/CustomizableClassHierarchy/ClassAndSubclasses.cpp
#include "pch.h"
#include "ClassAndSubclasses.h"

using namespace CustomizableClassHierarchy;
using namespace Platform;
using namespace Platform::Collections;
using namespace ReflectionHelper;

ClassAndSubclasses::ClassAndSubclasses(TypeInformation^ parent)
{
    this->Parent = parent;
    this->Subclasses = ref new Vector<ClassAndSubclasses^>();
}

File: 操作系统/Windows API/Windows 程序设计/第6版/CSharp/Chapter04/DependencyObjectClassHierarchy/DependencyObjectClassHierarchy/ClassAndSubclasses.cs
using System;
using System.Collections.Generic;

namespace DependencyObjectClassHierarchy
{
    class ClassAndSubclasses
    {
        public ClassAndSubclasses(Type parent)
        {
            this.Type = parent;
            this.Subclasses = new List<ClassAndSubclasses>();
        }

        public Type Type { protected set; get; }
        public List<ClassAndSubclasses> Subclasses { protected set; get; }
    }
}
